server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -251,6 +253,17 @@ private MessageWithId handleOutboxMessages(CreationMessageEntry entry, MailboxSe
         return new ValueWithId.MessageWithId(entry.getCreationId(), jmapMessage);
     }
 
+    private void assertUserIsSender(MailboxSession session, Optional<DraftEmailer> from) throws MailboxSendingNotAllowedException {
+        if (!from.map(DraftEmailer::getEmail)
+                .filter(Optional::isPresent)
```

Comment:
`.flatMap(OptionalUtils::toStream)` (or something like that, I don't have my IDE opened ...)